### ISSUE TITLE
installation problem on ubuntu 14.04

### ISSUE THREAD

**Issue by vitalije** [mikeseven/node-webgl#46]
Here is how I tried to install this package but with no success

```
vitalije@phenom:/tmp/t$ node --version
v4.2.3
vitalije@phenom:/tmp/t$ npm --version
2.14.7
vitalije@phenom:/tmp/t$ l
node_modules/
vitalije@phenom:/tmp/t$ rm -rf node_modules/
vitalije@phenom:/tmp/t$ node --version
v4.2.3
vitalije@phenom:/tmp/t$ npm --version
2.14.7
vitalije@phenom:/tmp/t$ npm install node-webgl
npm WARN engine node-glfw@0.4.4: wanted: {"node":">=0.6.5 <4.0.0"} (current: {"node":"4.2.3","npm":"2.14.7"})

> node-glfw@0.4.4 install /tmp/t/node_modules/node-webgl/node_modules/node-glfw
> node-gyp rebuild

make: Entering directory `/tmp/t/node_modules/node-webgl/node_modules/node-glfw/build'
  CXX(target) Release/obj.target/glfw/src/atb.o
  CXX(target) Release/obj.target/glfw/src/glfw.o
  SOLINK_MODULE(target) Release/obj.target/glfw.node
  COPY Release/glfw.node
make: Leaving directory `/tmp/t/node_modules/node-webgl/node_modules/node-glfw/build'

> node-webgl@0.5.5 install /tmp/t/node_modules/node-webgl
> node-gyp rebuild

make: Entering directory `/tmp/t/node_modules/node-webgl/build'
  CXX(target) Release/obj.target/webgl/src/bindings.o
  CXX(target) Release/obj.target/webgl/src/image.o
  CXX(target) Release/obj.target/webgl/src/webgl.o
  SOLINK_MODULE(target) Release/obj.target/webgl.node
  COPY Release/webgl.node
make: Leaving directory `/tmp/t/node_modules/node-webgl/build'
node-webgl@0.5.5 node_modules/node-webgl
├── bindings@1.2.1
├── nan@2.1.0
└── node-glfw@0.4.4
vitalije@phenom:/tmp/t$ node node_modules/node-webgl/test/lesson02.js 
node: symbol lookup error: /tmp/t/node_modules/node-webgl/node_modules/node-glfw/build/Release/glfw.node: undefined symbol: XIQueryVersion
vitalije@phenom:/tmp/t$
```

as one can see installation went with just a single warning, but whatever example I tried to run the result is like the above:
```
node: symbol lookup error: /tmp/t/node_modules/node-webgl/node_modules/node-glfw/build/Release/glfw.node: undefined symbol: XIQueryVersion
```
What am I doing wrong?

**Comment by mikeseven** [comment]
It would seem -lXi needs to be added to node-glfw.
Add it to bindings.gyp linker flags and rebuild node-glfw. Let me know if this solves running webgl samples.

Thanks

--mike



On Mon, Dec 7, 2015 at 7:28 AM -0800, "Виталије Милошевић" <notifications@github.com<mailto:notifications@github.com>> wrote:


Here is how I tried to install this package but with no success

vitalije@phenom:/tmp/t$ node --version
v4.2.3
vitalije@phenom:/tmp/t$ npm --version
2.14.7
vitalije@phenom:/tmp/t$ l
node_modules/
vitalije@phenom:/tmp/t$ rm -rf node_modules/
vitalije@phenom:/tmp/t$ node --version
v4.2.3
vitalije@phenom:/tmp/t$ npm --version
2.14.7
vitalije@phenom:/tmp/t$ npm install node-webgl
npm WARN engine node-glfw@0.4.4: wanted: {"node":">=0.6.5 <4.0.0"} (current: {"node":"4.2.3","npm":"2.14.7"})

> node-glfw@0.4.4 install /tmp/t/node_modules/node-webgl/node_modules/node-glfw
> node-gyp rebuild

make: Entering directory `/tmp/t/node_modules/node-webgl/node_modules/node-glfw/build'
  CXX(target) Release/obj.target/glfw/src/atb.o
  CXX(target) Release/obj.target/glfw/src/glfw.o
  SOLINK_MODULE(target) Release/obj.target/glfw.node
  COPY Release/glfw.node
make: Leaving directory `/tmp/t/node_modules/node-webgl/node_modules/node-glfw/build'

> node-webgl@0.5.5 install /tmp/t/node_modules/node-webgl
> node-gyp rebuild

make: Entering directory `/tmp/t/node_modules/node-webgl/build'
  CXX(target) Release/obj.target/webgl/src/bindings.o
  CXX(target) Release/obj.target/webgl/src/image.o
  CXX(target) Release/obj.target/webgl/src/webgl.o
  SOLINK_MODULE(target) Release/obj.target/webgl.node
  COPY Release/webgl.node
make: Leaving directory `/tmp/t/node_modules/node-webgl/build'
node-webgl@0.5.5 node_modules/node-webgl
??? bindings@1.2.1
??? nan@2.1.0
??? node-glfw@0.4.4
vitalije@phenom:/tmp/t$ node node_modules/node-webgl/test/lesson02.js
node: symbol lookup error: /tmp/t/node_modules/node-webgl/node_modules/node-glfw/build/Release/glfw.node: undefined symbol: XIQueryVersion
vitalije@phenom:/tmp/t$


as one can see installation went with just a single warning, but whatever example I tried to run the result is like the above:

node: symbol lookup error: /tmp/t/node_modules/node-webgl/node_modules/node-glfw/build/Release/glfw.node: undefined symbol: XIQueryVersion


What am I doing wrong?

-

**Comment by vitalije** [comment]
Thank you for fast reply. Unfortunately, applying your advice didn't solve the problem. Maybe I didn't apply it correctly. I have added ```'ldflags':['-lXi'] ``` to the bindings.gyp file under 
```OS=='linux'``` section, after ```'libraries':['-lfreeimage','-lGLEW','-lGL']``` and rebuild it, but the result is as before.

**Comment by mikeseven** [comment]
For node-glfw? Looks like you added it to jode-webgl. The problem is in new requirements for glfw3 on linux.

--mike



On Mon, Dec 7, 2015 at 10:23 AM -0800, "Виталије Милошевић" <notifications@github.com<mailto:notifications@github.com>> wrote:


Thank you for fast reply. Unfortunately, applying your advice didn't solve the problem. Maybe I didn't apply it correctly. I have added 'ldflags':['-lXi'] to the bindings.gyp file under
OS=='linux' section, after 'libraries':['-lfreeimage','-lGLEW','-lGL'] and rebuild it, but the result is as before.

-

**Comment by vitalije** [comment]
No I don't know about jode-webgl. Please can you explain me more clearly where should I add ```'-lXi'```. I have tried to put it in several places inside ```bindings.gyp``` file, but with no success.

**Comment by vitalije** [comment]
Ok. I got it. I should have added '-lXi' in node_modules/node-glfw/bindings.gyp and rebuild it, instead of adding it to the node-webgl/bindings.gyp.

**Comment by mikeseven** [comment]
Ok good to know it works. I'll update node-glfw files and readme.

Thanks!

--mike



On Mon, Dec 7, 2015 at 12:06 PM -0800, "Виталије Милошевић" <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #46<https://github.com/mikeseven/node-webgl/issues/46>.

-
